flake.nix

```diff
@@ -0,0 +1,56 @@
+{
+  description = "A Nix-flake-based Rust development environment";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/release-22.11";
+    flake-utils.url = "github:numtide/flake-utils";
+    rust-overlay.url = "github:oxalica/rust-overlay";
```

Comment:
we can just use rust from nixpkgs

Reply:
with the rust-overlay there is much greater control over what version of rust is utillized. For packaging I'd consider using rust from nixpkgs, but for developing, sometimes is necessary to quickly choose another version.

Reply:
rust-overlay is a large dependency, and instead having to manually edit files, it's probably better to use ad-hoc shells from the command line instead for quickly choosing different versions of rust, which is probably not something that needs to be done very often